pycargoebuild/__main__.py

```diff
@@ -71,13 +72,22 @@ def main(prog_name: str, *argv: str) -> int:
     load_license_mapping(args.license_mapping)
     args.license_mapping.close()
 
+    def locate_cargo_lock_file(directory: Path) -> io.BytesIO:
+        for current_dir in (directory.resolve() / "Cargo.lock" ).parents:
```

Comment:
```suggestion
        for current_dir in (directory.resolve() / "Cargo.lock").parents:
```

Reply:
That said, feel free not to want to do that but I'm not a big fan of realpath. I'd personally prefer just appending `../`s until you hit the filesystem root (i.e. `.samefile(directory.root)`).

Reply:
I have no strong opinion here, but I see some benefits in appending `..` over `realpath`, however `directory.root` does not always return the root (`/`), it returns empty path for relative paths. Then the `current_dir.samefile(directory.root)` raises `[Errno 2] No such file or directory: ''` exception in such situations.

Anyway, I can rewrite it to `..` appending, but the condition needs to be changed to something else like `current_dir.samefile(Path('/'))` if you have nothing against it.

Reply:
Maybe `current_dir.absolute().root`? That should be safer for Windows.

Reply:
Nice, this does the job, thanks!